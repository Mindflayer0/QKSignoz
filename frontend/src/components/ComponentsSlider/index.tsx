import React from 'react';

import menuItems from './menuItems';
import { Card, Container, Text } from './styles';

const DashboardGraphSlider = (): JSX.Element => {
	return (
		<Container>
			{menuItems.map(({ name, Icon }) => (
				<Card key={name} draggable>
					<Icon />
					<Text>{name}</Text>
				</Card>
			))}
		</Container>
	);
};

export default DashboardGraphSlider;
