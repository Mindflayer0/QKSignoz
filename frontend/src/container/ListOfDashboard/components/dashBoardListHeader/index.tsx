import { PlusOutlined } from '@ant-design/icons';
import { Dropdown, MenuProps, Row, Typography } from 'antd';
import TextToolTip from 'components/TextToolTip';
import { NewDashboardState } from 'container/ListOfDashboard';
import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';

import { ButtonContainer, NewDashboardButton } from '../../styles';
import { getDashBoardText, getMenuItems } from './helpers';

function DashBoardListHeader({
	isDashboardListLoading,
	newDashboard,
	newDashboardState,
	createNewDashboard,
	onModalHandler,
	onNewDashboardHandler,
}: Props): JSX.Element {
	const { t } = useTranslation('dashboard');

	const menu: MenuProps = useMemo(
		() => ({
			items: getMenuItems({
				createNewDashboard,
				isDashboardListLoading,
				onModalHandler,
				onNewDashboardHandler,
				t,
			}),
		}),
		[
			createNewDashboard,
			isDashboardListLoading,
			onModalHandler,
			onNewDashboardHandler,
			t,
		],
	);

	return (
		<Row justify="space-between">
			<Typography>Dashboard List</Typography>

			<ButtonContainer>
				<TextToolTip
					text="More details on how to create dashboards"
					url="https:signoz.io/docs/userguide/dashboards"
				/>
				{newDashboard && (
					<Dropdown
						disabled={isDashboardListLoading}
						trigger={['click']}
						menu={menu}
					>
						<NewDashboardButton
							icon={<PlusOutlined />}
							type="primary"
							loading={newDashboardState.loading}
							danger={newDashboardState.error}
						>
							{getDashBoardText(newDashboardState)}
						</NewDashboardButton>
					</Dropdown>
				)}
			</ButtonContainer>
		</Row>
	);
}

interface Props {
	newDashboard: boolean;
	isDashboardListLoading: boolean;
	newDashboardState: NewDashboardState;
	createNewDashboard: boolean;
	onNewDashboardHandler: () => Promise<void>;
	onModalHandler: (uploadedGrafana: boolean) => void;
}

export default DashBoardListHeader;
