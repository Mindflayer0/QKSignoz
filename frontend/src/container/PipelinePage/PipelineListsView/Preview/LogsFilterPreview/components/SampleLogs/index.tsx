import './styles.scss';

import { Time } from 'container/TopNav/DateTimeSelection/config';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';

import LogsList from '../../../components/LogsList';
import useSampleLogs from '../../../hooks/useSampleLogs';

function SampleLogs({ filter, timeInterval }: SampleLogsProps): JSX.Element {
	const { isLoading, isError, logs } = useSampleLogs({
		filter,
		timeInterval,
		count: 5,
	});

	if (isError) {
		return (
			<div className="sample-logs-notice-container">
				could not fetch logs for filter
			</div>
		);
	}

	if (isLoading) {
		return <div className="sample-logs-notice-container">Loading...</div>;
	}

	if ((filter?.items?.length || 0) < 1) {
		return (
			<div className="sample-logs-notice-container">Please select a filter</div>
		);
	}

	if (logs.length < 1) {
		return <div className="sample-logs-notice-container">No logs found</div>;
	}

	return <LogsList logs={logs} />;
}

interface SampleLogsProps {
	filter: TagFilter;
	timeInterval: Time;
}

export default SampleLogs;
