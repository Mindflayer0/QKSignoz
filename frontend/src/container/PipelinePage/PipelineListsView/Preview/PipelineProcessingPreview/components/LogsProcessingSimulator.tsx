import { Button } from 'antd';
import { useState } from 'react';
import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

function LogsProcessingSimulator({
	inputLogs,
	pipeline,
}: LogsProcessingSimulatorProps): JSX.Element {
	const [showResult, setShowResult] = useState<boolean>(false);
	const simulate = (): void => setShowResult(true);

	if (!showResult) {
		return (
			<div>
				<Button type="primary" onClick={simulate}>
					Simulate Processing
				</Button>
			</div>
		);
	}

	return <div>{JSON.stringify({ inputLogs, pipeline })}</div>;
}

export interface LogsProcessingSimulatorProps {
	inputLogs: ILog[];
	pipeline: PipelineData;
}

export default LogsProcessingSimulator;
