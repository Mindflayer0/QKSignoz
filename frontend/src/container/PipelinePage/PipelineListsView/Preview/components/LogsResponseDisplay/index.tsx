import { LogsResponse } from '../../types';
import LogsList from '../LogsList';

function LogsResponseDisplay({
	response,
}: LogsResponseDisplayProps): JSX.Element {
	const { isLoading, isError, logs } = response;

	if (isError) {
		return (
			<div className="sample-logs-notice-container">There was an error.</div>
		);
	}

	if (isLoading) {
		return <div className="sample-logs-notice-container">Loading...</div>;
	}

	if (logs.length < 1) {
		return <div className="sample-logs-notice-container">No logs found</div>;
	}

	return <LogsList logs={logs} />;
}

export interface LogsResponseDisplayProps {
	response: LogsResponse;
}

export default LogsResponseDisplay;
