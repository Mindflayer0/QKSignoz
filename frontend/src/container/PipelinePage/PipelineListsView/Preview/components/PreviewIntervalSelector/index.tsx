import './styles.scss';

import { Select } from 'antd';
import {
	initialFilters,
	initialQueriesMap,
	PANEL_TYPES,
} from 'constants/queryBuilder';
import {
	RelativeDurationOptions,
	Time,
} from 'container/TopNav/DateTimeSelection/config';
import { useGetQueryRange } from 'hooks/queryBuilder/useGetQueryRange';
import { cloneDeep } from 'lodash-es';
import { useMemo } from 'react';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';
import { LogsAggregatorOperator } from 'types/common/queryBuilder';

function MatchedLogsCount({
	filter,
	timeInterval,
}: MatchedLogsCountProps): JSX.Element {
	const query = useMemo(() => {
		const q = cloneDeep(initialQueriesMap.logs);
		q.builder.queryData[0] = {
			...q.builder.queryData[0],
			filters: filter || initialFilters,
			aggregateOperator: LogsAggregatorOperator.COUNT,
		};
		return q;
	}, [filter]);

	const result = useGetQueryRange({
		graphType: PANEL_TYPES.TABLE,
		query,
		selectedTime: 'GLOBAL_TIME',
		globalSelectedInterval: timeInterval,
	});

	if (result.isFetched) {
		const count =
			result?.data?.payload?.data?.newResult?.data?.result?.[0]?.series?.[0]
				?.values?.[0]?.value;
		return (
			<div className="logs-filter-preview-matched-logs-count">
				{count} matches in
			</div>
		);
	}
	return <div />;
}

interface MatchedLogsCountProps {
	filter: TagFilter;
	timeInterval: Time;
}

function PreviewIntervalSelector({
	previewFilter,
	value,
	onChange,
}: PreviewIntervalSelectorProps): JSX.Element {
	const isEmptyFilter = (previewFilter?.items?.length || 0) < 1;

	return (
		<div className="logs-filter-preview-time-interval-summary">
			{!isEmptyFilter && (
				<MatchedLogsCount filter={previewFilter} timeInterval={value} />
			)}
			<div>
				<Select
					onSelect={(value: unknown): void => onChange(value as Time)}
					value={value}
				>
					{RelativeDurationOptions.map(({ value, label }) => (
						<Select.Option key={value + label} value={value}>
							{label}
						</Select.Option>
					))}
				</Select>
			</div>
		</div>
	);
}

interface PreviewIntervalSelectorProps {
	value: Time;
	onChange: (interval: Time) => void;
	previewFilter: TagFilter;
}

export default PreviewIntervalSelector;
