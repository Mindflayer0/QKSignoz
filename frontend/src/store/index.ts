import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
