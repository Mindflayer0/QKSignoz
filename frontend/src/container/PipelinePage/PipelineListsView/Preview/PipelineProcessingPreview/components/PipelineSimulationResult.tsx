import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

import LogsResponseDisplay from '../../components/LogsResponseDisplay';
import usePipelinePreview from '../../hooks/usePipelinePreview';

function PipelineSimulationResult({
	inputLogs,
	pipeline,
}: PipelineSimulationResultProps): JSX.Element {
	const simulationResult = usePipelinePreview({ pipeline, logs: inputLogs });

	return <LogsResponseDisplay response={simulationResult} />;
}

export interface PipelineSimulationResultProps {
	inputLogs: ILog[];
	pipeline: PipelineData;
}

export default PipelineSimulationResult;
