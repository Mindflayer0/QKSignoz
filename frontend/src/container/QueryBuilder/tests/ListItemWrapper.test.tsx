import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { ListItemWrapper } from '../components';

describe('QueryBuilder: List Item Wrapper', () => {
	it('it should render ListItemWrapper', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<ListItemWrapper onDelete={jest.fn()}>List Item Wrapper</ListItemWrapper>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
