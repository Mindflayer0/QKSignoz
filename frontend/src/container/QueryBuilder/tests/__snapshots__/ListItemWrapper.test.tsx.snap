// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder: List Item Wrapper it should render ListItemWrapper 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: 0.5rem;
  right: 0.9375rem;
  z-index: 1;
  cursor: pointer;
  opacity: 0.45;
  width: 1.3125rem;
  height: 1.3125rem;
}

.c1 svg {
  width: 100%;
  height: 100%;
}

.c0 {
  padding-right: 3rem;
}

<div
    class="ant-row c0 css-dev-only-do-not-override-1i536d8"
    style="margin-top: -7.5px; margin-bottom: -7.5px;"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle c1"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
        />
        <path
          d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
      </svg>
    </span>
    List Item Wrapper
  </div>
</DocumentFragment>
`;
